site/content/en/references/kustomize/kustomization/buildmetadata/_index.md

```diff
@@ -0,0 +1,308 @@
+- deployment.yaml
+
+buildMetadata: [originAnnotations]
+```
+
+would produce a resource with an annotation like the following:
+
+```yaml
+config.kubernetes.io/origin: |
+  path: deployment.yaml
+```
+
+##### Local custom generator
+A kustomization such as the following:
+
+```yaml
+generators:
+- generator.yaml
+
+buildMetadata: [originAnnotations]
+```
+
+would produce a resource with an annotation like the following:
+
+```yaml
+config.kubernetes.io/origin: |
+  configuredIn: generator.yaml
+  configuredBy: 
+    kind: MyGenerator
+    apiVersion: v1
+    name: generator
+```
+
+##### Remote builtin generator
+We have a top-level kustomization such as the following:
+
+```yaml
+resources:
+- github.com/examplerepo/?ref=v1.0.6
+
+buildMetadata: [originAnnotations]
+```
+
+which uses `github.com/examplerepo/?ref=v1.0.6` as a remote base. This remote base has the following kustomization
+defined in `github.com/examplerepo/kustomization.yaml`:
+
+```yaml
+configMapGenerator:
+- name: my-java-server-env-vars
+  literals:
+  - JAVA_HOME=/opt/java/jdk
+  - JAVA_TOOL_OPTIONS=-agentlib:hprof
+```
+
+Running `kustomize build` on the top-level kustomization would produce the following output:
+
+```yaml
+apiVersion: v1
+data:
+  JAVA_HOME: /opt/java/jdk
+  JAVA_TOOL_OPTIONS: -agentlib:hprof
+kind: ConfigMap
+metadata:
+  name: my-java-server-env-vars-44k658k8gk
+  annotations:
+    config.kubernetes.io/origin: |
+      ref: v1.0.6 
+      repo: github.com/examplerepo 
+      configuredIn: kustomization.yaml
+      configuredBy: 
+        kind: ConfigMapGenerator
+        apiVersion: builtin 
+```
+
+### Transformer Annotations [Alpha]
+
+To retain the information about what transformers have acted on each resource, you can add the
+`transformerAnnotations` option to the `buildMetadata` field of the kustomization:
+
+```yaml
+buildMetadata: [transformerAnnotations]
+```
+
+When the `transformerAnnotations` option is set, kustomize will add annotations with information about what transformers 
+have acted on each resource. Transformers can be invoked either through various fields in the kustomization file 
+(e.g. the `replacements` field will invoke the ReplacementTransformer), or through the `transformers` field.
+
+The annotation key for transformer annotations will be `config.kubernetes.io/transformations`, which will contain a list of
+transformer data:
+
+```yaml
+config.kubernetes.io/transformations: |
+- ref: v0.0.0 # If from a remote transformer, the ref of the repo URL
+  repo: http://github.com/examplerepo # If from a remote transformer, the repo source
+  configuredIn: path/to/transformerconfig # The path to the transformer config
+  configuredBy: # The ObjectReference of the transformer config
+    kind: Transformer
+    apiVersion: builtin 
+    name: foo
+    namespace: default
+```
+
+While this field is in alpha, it will receive the `alpha` prefix, so you will see the annotation key 
+`alpha.config.kubernetes.io/transformations` instead. 
```

Comment:
I think I asked for this, but did we decide that we are not guaranteeing that the annotation content will be stable during alpha? If so, we should mention that.